case_studies/galaxy_clustering/prior.py

```diff
@@ -0,0 +1,338 @@
+                n_galaxy_cluster.append(0)
+        return n_galaxy_cluster
+
+    def _sample_center(self):
+        x_coords = np.random.uniform(self.width * 0.2, self.width * 0.8, self.size)
+        y_coords = np.random.uniform(self.height * 0.2, self.height * 0.8, self.size)
+        return np.vstack((x_coords, y_coords)).T
+
+    # Spherical Sampling Method: https://stackoverflow.com/a/5408843
+    def _sample_cluster_locs(self, center_samples, radius_samples, n_galaxy_cluster):
+        galaxy_locs_cluster = []
+        for i in range(self.size):
+            center_x, center_y = center_samples[i]
+            samples = []
+            while len(samples) < int(n_galaxy_cluster[i]):
+                phi = np.random.uniform(0, 2 * np.pi, 1)
+                sintheta = np.random.uniform(-1, 1, 1)
+                u = np.random.uniform(0, 1, 1)
+                r_in_sphere = radius_samples[i] * np.cbrt(u)
+                shift_x = r_in_sphere * sintheta * np.cos(phi)
+                shift_y = r_in_sphere * sintheta * np.sin(phi)
+                sampled_x = float(center_x + shift_x)
+                sampled_y = float(center_y + shift_y)
+                if 0 <= sampled_x < self.width and 0 <= sampled_y < self.height:
+                    samples.append([sampled_x, sampled_y])
+
+            galaxy_locs_cluster.append(samples)
+        return galaxy_locs_cluster
+
+    def _sample_n_galaxy(self):
+        return np.random.poisson(self.mean_sources * self.width * self.height / 49, self.size)
+
+    def _sample_galaxy_locs(self, n_galaxy):
+        galaxy_locs = []
+        for i in range(self.size):
+            x = np.random.uniform(0, self.width, n_galaxy[i])
+            y = np.random.uniform(0, self.height, n_galaxy[i])
+            galaxy_locs.append(np.column_stack((x, y)))
+        return galaxy_locs
+
+    def cartesian2geo(self, coordinates, pixel_scale=0.2, image_offset=(2499.5, 2499.5)):
+        sky_center = (self.ra_cen, self.dec_cen)
+        geo_coordinates = []
+        for coord_i in coordinates:
+            temp = []
+            for coord_ij in coord_i:
+                ra = (coord_ij[0] - image_offset[0]) * pixel_scale / (60 * 60) + sky_center[0]
+                dec = (coord_ij[1] - image_offset[1]) * pixel_scale / (60 * 60) + sky_center[1]
+                temp.append((ra, dec))
+            geo_coordinates.append(temp)
+        return geo_coordinates
+
+    def _sample_tsize(self, flux_samples):
+        t_size_samples = []
+        for i in range(self.size):
+            t_size_samples.append(self.tsize_poly(flux_samples[i]))
+        return t_size_samples
+
+    def _sample_redshift_bg(self):
+        redshift_bins = np.linspace(0.01, 7, 100)
+        redshift_pdf = [self._redshift_distribution(z) for z in redshift_bins]
+        redshift_pdf = redshift_pdf / np.sum(redshift_pdf)
+        return (
+            np.random.choice(np.linspace(0.01, 7, 100), p=redshift_pdf)
+            + (np.random.random()) * 0.07
```

Comment:
I'd be clearer if we used some intermediate variables here, rather than having a multi-line return statement